### PR TITLE
sd-id128: several followups

### PR DESCRIPTION
**Pull request by yuwata**

### CODE REVIEW

File: src/libsystemd/sd-id128/id128-util.c [hunk 1]
```diff
@@ -54,7 +55,7 @@ int id128_read_fd(int fd, Id128FormatFlag f, sd_id128_t *ret) {
          * This returns the following:
          *     -ENOMEDIUM: an empty string,
          *     -ENOPKG:    "uninitialized" or "uninitialized\n",
-         *     -EINVAL:    other invalid strings. */
+         *     -EIO:       other invalid strings. */
```
Comment (keszybz) [inline src/libsystemd/sd-id128/id128-util.c]: I know this is very close to bike-shedding, but I think we shouldn't use EIO. EIO is for "input/output error", i.e. the error you'd get if the disk or file system is borked, and this error code could be returned by the underlying read/write functions. It'd be better to use something unambiguous, like EUCLEAN.

File: man/sd_id128_get_machine.xml [hunk 2]
```diff
@@ -94,10 +94,20 @@
     has properties similar to the machine ID during that time.</para>
 
     <para><function>sd_id128_get_invocation()</function> returns the invocation ID of the currently executed
-    service. In its current implementation, this reads and parses the <varname>$INVOCATION_ID</varname> environment
-    variable that the service manager sets when activating a service, see
-    <citerefentry><refentrytitle>systemd.exec</refentrytitle><manvolnum>5</manvolnum></citerefentry> for details. The
-    ID is cached internally. In future a different mechanism to determine the invocation ID may be added.</para>
+    service. In its current implementation, this tries to read and parse the followings:
```
Comment (keszybz) [inline man/sd_id128_get_machine.xml]: "following"

File: man/sd_id128_get_machine.xml [hunk 3]
```diff
@@ -94,10 +94,20 @@
...
+    <itemizedlist>
+      <listitem>
+        <para>The <varname>$INVOCATION_ID</varname> environment variable that the service manager sets when
+        activating a service.</para>
+      </listitem>
+      <listitem>
+        <para>The per-service keryring that the system service manager sets when activating a service.
```
Comment (keszybz) [inline man/sd_id128_get_machine.xml]: "keyring"

It's not parsing a keyring, but an entry in the keyring. So this needs rewording.

**Review by keszybz — CHANGES_REQUESTED** [review]
> sd-id128: sd_id128_get_machine() and friend can call with NULL 

When the commit title is phrased like this, it sounds like a statement about *status quo*. Please say something like "allow sd_id128_get_machine() and friend to be called with NULL", so that it's unambiguous that you're changing this.

**Comment by yuwata** [comment]
@keszybz Thank you for the review.
- the commit titles and messages are updated,
- `-EIO` -> `-EUCLEAN`
- updated man page.

The change is not small, hence, re-setting the 'please-review' tag. PTAL!

**Pull request merged**